### PR TITLE
return signed offset from alg GetDetectorOffsets

### PR DESCRIPTION
**Pull request by stonecoldhughes**
**Description of work.**

Add "signed" mode to GetDetectorOffsets. Update documentation. Still in progress.

PR ---> ornl-next: https://github.com/mantidproject/mantid/pull/31722

**To test:**

> cd build/bin;
> ./AlgorithmsTest GetDetectorOffsetsTest

Fixes https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/263
 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/GetDetectorOffsets-v1.rst [hunk 1]
```diff
@@ -18,10 +18,22 @@ defines the offset from the reference spectrum.
...
+:math:`offset = -1*peakCentre*step/(dreference+PeakCentre*step)`
+
+Absolute (offset relative to ideal position):
+
+:math:`offset = -1*peakCentre*step/(dreference+PeakCentre*step) + (dideal - dreference) / dreference`
+
+
+Signed (offset in raw number of bins):
+
+:math:`offset = peak_center*-1`
```
Comment (rosswhitfield) [inline docs/source/algorithms/GetDetectorOffsets-v1.rst:36]: This isn't rendering correctly
2021-06-04-160824_266x42_scrot

File: docs/source/algorithms/GetDetectorOffsets-v1.rst [hunk 2]
```diff
@@ -18,10 +18,22 @@ defines the offset from the reference spectrum.
...
-to calculate the centre of the fitted peak, and the offset is then
-calculated as:
+to calculate the centre of the fitted peak. Using the fitted peak center,
+there are 3 options for calculating offset, depending on the choice for the "OffsetMode"
+input parameter:
 
-:math:`-peakCentre*step/(dreference+PeakCentre*step)`
+Relative (offset relative to reference position):
+
+:math:`offset = -1*peakCentre*step/(dreference+PeakCentre*step)`
```
Comment (rosswhitfield) [inline docs/source/algorithms/GetDetectorOffsets-v1.rst:27]: You probably don't need to include the `-1*`, just do `-peakCentre*...`. Same for the other equations.

File: Framework/Algorithms/src/GetDetectorOffsets.cpp [hunk 3]
```diff
@@ -204,16 +212,33 @@ double GetDetectorOffsets::fitSpectra(const int64_t s, bool isAbsolbute) {
...
   // std::vector<double> params = fit_alg->getProperty("Parameters");
   API::IFunction_sptr function = fit_alg->getProperty("Function");
+
   double offset = function->getParameter(3); // params[3]; // f1.PeakCentre
-  offset = -1. * offset * m_step / (m_dreference + offset * m_step);
-  // factor := factor * (1+offset) for d-spacemap conversion so factor cannot be
-  // negative
 
-  if (isAbsolbute) {
+  if (mode == offset_mode::signed_offset) {
```
Comment (rosswhitfield) [inline Framework/Algorithms/src/GetDetectorOffsets.cpp:218]: Can you add a test for this new mode https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/test/GetDetectorOffsetsTest.h

**Review by rosswhitfield — CHANGES_REQUESTED** [review]

**Review by rosswhitfield — APPROVED** [review]

**Comment by stonecoldhughes** [comment]
@mantidproject/gatekeepers this is ready for merge! Thanks

**Pull request merged**